lib/elixir/lib/macro/env.ex

```diff
@@ -235,6 +235,31 @@ defmodule Macro.Env do
     f ++ m
   end
 
+  @doc """
+  Returns the names of any aliases for the given module or atom.
+
+  ## Examples
+
+      iex> alias Foo.Bar
+      iex> Bar
+      Foo.Bar
+      iex> Macro.Env.lookup_alias_as(__ENV__, Foo.Bar)
+      [:Bar]
+      iex> alias Foo.Bar, as: Baz
+      iex> Baz
+      Foo.Bar
+      iex> Macro.Env.lookup_alias_as(__ENV__, Foo.Bar)
+      [:Bar, :Baz]
+      iex> Macro.Env.lookup_alias_as(__ENV__, Unknown)
+      []
+
+  """
+  @doc since: "1.15.0"
+  @spec lookup_alias_as(t, atom) :: [atom]
+  def lookup_alias_as(%{__struct__: Macro.Env, aliases: aliases}, atom) when is_atom(atom) do
+    for {name, ^atom} <- aliases, do: String.to_atom(hd(Module.split(name)))
```

Comment:
Sorry, I should have noticed this earlier, but I am thinking we should return the `name` as is. The reason is because it is immediately useful. For example, in Kino.VegaLite, we would do this:

    __ENV__ |> Macro.Env.lookup_alias_as(VegaLite) |> List.first(VegaLite)

However, if we preprocess, like we do here, we would have to postprocess to put it back to its original value. Plus the `:as` is enforce to always be an `alias`, it can never be a bare atom like `:Bar`.

Reply:
To be clear:

```suggestion
    for {name, ^atom} <- aliases, do: name
```

:)

Reply:
Sounds good! This is what I first had but for some reason, I got it in my head that it wasn't possible because the doctests were failing when I asserted the result was `[Bar]` etc., but I just needed to be explicit about `[:"Elixir.Bar"]` to avoid the alias resolving.

I can squash the commit that changes this if you'd like after we're done reviewing.

Reply:
You can write them as `Elixir.Bar` and that will disable any resolution!